### PR TITLE
Docker dev ux

### PR DESCRIPTION
**Pull request by madaster97**
I added some dev-specific dockerfiles and a docker-compose file that connects all the right folders as bind mounts and volumes.

This makes for a better dev UX since the dev dockerfiles have less layers than the main Dockerfile for the repo. I just run `npm ci` and copy the right source files in before running the corresponding `dev`/`dev-ui` script. I also have the local directory's `src` folder mapped to each of the containers' so watchers can pick up on changes in either container OR the host.

### CODE REVIEW

**Comment by jmandel** [comment]
I managed not to get notifications on this - sorry for the slow response! This is great.

File: README.md [hunk 1]
```diff
@@ -47,5 +47,16 @@ This demo should work wiht Node.js 15 (current LTS) as well as Node.js 13. See `
...
 
+## Run dev-only containers with watchers in Docker-Compose
+    export USER=$(id -u) 
+    docker-compose --env-file ./compose.env up
+   
+You can use the docker-compose.yaml file to spin up two dev containers with watchers, one for the UI and one for the server.
+
+### Note the following:
+1. Both containers have their `src` directories bind-mounted to the local directory's `src` folder. Any changes made in the `dev` container (or the host) will propagate to both containers + host and be registered by the watchers. This is helpful since you can, for instance, launch programs inside the `dev` container and utilize dev dependencies without needing to ever install them locally.
+2. Both containers have their `dist` folders mounted to a named volume. This means the parcel watcher in the `dev-ui` container can write changes that are accessible by the `dev` container. Note the `dev-ui` container needs `root` priviliges for this. See this [issue](https://github.com/moby/moby/issues/2259) for details.
```
Comment (jmandel) [inline README.md:58]: Would it be possible to avoid this root permissions issue by mounting a `dist` directory from the host into the dev containers?

Comment (madaster97) [reply README.md:58]: Yeah we can get around this that way. That's probably the simplest answer. See this [commit](https://github.com/microsoft-healthcare-madison/health-wallet-demo/commit/38b456cc0a14955d280b19f754181932ef58b868) for the fix.

Comment (jmandel) [reply README.md:58]: Perfect. I've created a PR from you branch so you don't need to go through the trouble of re-opening.

**Pull request closed without merging**